services/app-web/src/contexts/S3Context.tsx

```
@@ -51,9 +53,17 @@ const useS3 = (): S3ContextT => {
         if (isS3Error(s3Key)) {
             const error = new Error(`Error in S3: ${file.name}`)
             recordJSException(error)
+            // s3 file upload failing could be due to IDM session timeout
+            // double check the user still has their session, if not, logout to update the React state with their login status
+            try {
+                await checkAuth()
```

why does checkAuth throw instead of return an error?